### PR TITLE
feat(funnels): breakdown attributions FE

### PR DESCRIPTION
**Pull request by liyiy**
## Changes

I think we need to pass `breakdown_attribution_type` and `value` from the backend to `filters` in the FE? 😁 @neilkakkar 

Screen Shot 2022-06-14 at 10 26 03 AM
Screen Shot 2022-06-14 at 10 26 21 AM

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

File: frontend/src/scenes/insights/InsightTabs/FunnelTab/FunnelTab.tsx [hunk 1]
```diff
@@ -155,6 +167,62 @@ export function FunnelTab(): JSX.Element {
...
+                                                                      if (value) {
+                                                                          setFilters({
+                                                                              breakdown_attribution_type:
+                                                                                  BreakdownAttributionType.Step,
+                                                                              breakdown_attribution_value: value,
+                                                                          })
+                                                                      }
+                                                                  }}
+                                                                  placeholder={`Step ${
+                                                                      filters.breakdown_attribution_value || 1
```
Comment (neilkakkar) [inline frontend/src/scenes/insights/InsightTabs/FunnelTab/FunnelTab.tsx:203]: quick flyby that `breakdown_attribution_value` is zero indexed while steps are 1 indexed 😅

File: frontend/src/types.ts [hunk 2]
```diff
@@ -1308,6 +1310,13 @@ export interface FlattenedFunnelStepByBreakdown {
     significant?: boolean
 }
 
+export enum BreakdownAttributionType {
+    FirstTouch = 'first_touch',
+    LastTouch = 'last_touch',
+    AnyStep = 'all_events',
```
Comment (neilkakkar) [inline frontend/src/types.ts:1316]: any step for unordered is same as step = 0. 

All_events is for the 4th option, where we want the property to be present on all events.

File: frontend/src/scenes/insights/InsightTabs/FunnelTab/FunnelTab.tsx (hunk 1 above)
Comment (liyiy) [reply frontend/src/scenes/insights/InsightTabs/FunnelTab/FunnelTab.tsx:203]: :P

Comment (liyiy) [reply frontend/src/scenes/insights/InsightTabs/FunnelTab/FunnelTab.tsx:203]: wait is value also 1 indexed?

File: frontend/src/types.ts (hunk 2 above)
Comment (liyiy) [reply frontend/src/types.ts:1316]: so you want it to be all 4 options available? So not like the mocks where it's only 3 at all times then? https://www.figma.com/file/gQBj9YnNgD8YW4nBwCVLZf/PostHog-App?node-id=11313%3A49272

File: frontend/src/scenes/insights/InsightTabs/FunnelTab/FunnelTab.tsx (hunk 1 above)
Comment (neilkakkar) [reply frontend/src/scenes/insights/InsightTabs/FunnelTab/FunnelTab.tsx:203]: what do you mean? 😅 .

`breakdown_attribution_value` follows the 'order' in the entity definition. If order = 0, if you want step 1, value = 0 too.

If you want step = 3, value = 2, order = 2

File: frontend/src/types.ts (hunk 2 above)
Comment (liyiy) [reply frontend/src/types.ts:1316]: I can update it but if it's also faster for you to make these tweaks yourself feel free to! Either works for me :)

**Comment by neilkakkar** [comment]
Hey Li!

> I think we need to pass breakdown_attribution_type and value from the backend to filters in the FE?

This is already happening. I just ran locally to confirm, and all API requests are happening properly, just required a small fix.

---

What's missing:

1. [x] Adhering to the updated mocks. I added all steps, but haven't done the design changes 😅 
2. [x] Populating the `LemonSelect` with the value coming from the filter, when opening a funnel for the first time.
3. [x] Instead of having to select an option, we can always default to First Touchpoint when creating a new funnel.

**Comment by neilkakkar** [comment]
code-wise looks good to me!

Added Chris for a design review. Here's how the latest looks like, @clarkus :

**Review by clarkus — APPROVED** [review]
This looks good to me, but I think it's problematic that you set attribution before you set funnel ordering. That's bigger than this issue, but ideally funnel ordering would be set prior to attribution types because attribution type is affected by funnel ordering. I'm working on some ideas for this distinct from this effort. I'll file an issue for that work once it's ready.

**Pull request merged**